src/api/clients/BaseApiClient.ts

```diff
@@ -0,0 +1,99 @@
+import axios, { AxiosError } from "axios";
+import { ApiError } from "../types/ApiResponses";
+import { HttpMethod } from "../types/Enums";
+
+export abstract class BaseApiClient{
+
+    authToken: string;
+
+    constructor(authToken: string){
+        this.authToken = authToken;
+    }
```

Comment:
What do you think about including the url to this class too? Seems like this base Api Client will always be used with the URL and the inherited classes could re-use the constructor.